File: 백준/Silver/18870. 좌표 압축/좌표 압축.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        StringTokenizer token = new StringTokenizer(br.readLine(), " ");
        ArrayList<Integer> numList = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            numList.add(Integer.parseInt(token.nextToken()));
        }

        Set<Integer> numSet = new TreeSet<>(numList);
        TreeMap<Integer, Integer> dict = new TreeMap<>();
        int index = 0;
        for (Integer num : numSet) {
            dict.put(num, index);
            index++;
        }

        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < numList.size(); i++) {
            sb.append(dict.get(numList.get(i))).append(' ');
        }

        System.out.println(sb);
    }
}

File: 백준/Silver/18870. 좌표 압축/좌표 압축.py
import sys
input = sys.stdin.readline

n = int(input())
num_list = list(map(int, input().split()))
num_set = sorted(set(num_list))
dict = {i:index for index, i in enumerate(num_set)}

ans = []
for i in range(len(num_list)):
    ans.append(dict[num_list[i]])

print(*ans)